specs/agents/tracing-spans-discarding.md

```diff
@@ -0,0 +1,128 @@
+# Discarding fast spans
+
+Currently,
+the experience of our APM solution is not ideal when monitoring an application that ends up generating thousands of spans per request.
+
+We currently have a default limit of 500 spans in the agents and 1000 spans in the UI to prevent overwhelming the system with too many spans.
+However,
+that just cuts off at a certain point without considering which spans are important and which are not.
+
+## `span_min_duration` configuration
+
+Sets the minimum duration of spans.
+Spans that execute faster than this threshold are attempted to be discarded.
+
+The attempt fails if they lead up to a span that can't be discarded.
+Spans that propagate the trace context to downstream services,
+such as outgoing HTTP requests,
+can't be discarded.
+Additionally, spans that lead to an error or that may be a parent of an async operation can't be discarded.
+
```

Comment:
> that may be a parent of an async operation can't be discarded.

Repeating what was mentioned elsewhere -- this requirement means that for the Node.js Agent "in the real world" that no spans will be eligible for discarding. 

The culture around Node.js heavily encourages the use of asynchronous programming patterns -- specifically asynchronous callback APIs and Promise based APIs.  For example -- the typical handling of a HTTP(s) request in Node.js will look like

1. Register a callback function to handle a specific route/url
2. This callback makes a request to the application's data layer (db query, orm, etc) for some data, which occurs asynchronously, and registers a second callback function
3. This route handling callback finishes execution. 
4. When the asynchronous request for data finishes, node calls the second callback. This callback typically handles sending the HTTP response

While it's possible to construct a route handling callback that works synchronously in Node.js -- it's not very common in the real world. 

@nehaduggal (since @alex-fedotyev is currently on leave) -- the above means that the `span_min_duration` feature, as speced, wouldn't help any Node.js users with their too many spans problems.